schema_editor/app/scripts/views/recordtype/add-controller.js

```diff
@@ -31,6 +31,7 @@
             var schema = Schemas.JsonObject();
             schema = Schemas.addVersion4Declaration(schema); // Make root object a "real" JSON-Schema
             var definition = Schemas.JsonObject();
+            definition.details = true; // Flag denoting that this is the 'details' definition
```

Comment:
Did you confirm that this survives when the Details schema is edited using the schema builder?